### DISCUSSION TITLE
Extensions: allow plugins to return more than one `extensionLink` from `configure`

### DISCUSSION THREAD

**Discussion by sd2k** [grafana/grafana#66294] [Plugins]
The signature of the `configure` field of `PluginExtensionConfig` currently looks like this:

```
    // (Optional) A function that can be used to configure the extension dynamically based on the extension point's context
    configure?: (
      context?: Readonly<Context>
    ) => Partial<{ title: string; description: string } & ExtraProps> | undefined;
```

This conveniently allows plugins to customise the extension link based on the context provided by the context point. However, there may be cases where the plugin wants to offer multiple actions for the current context. Concrete use cases include the alerting and Machine Learning plugins, which (on a dashboard panel extension point) may want to offer the ability to create an alert (or respectively ML forecast) for each query in the panel context.

This would be simple from a plugin perspective if the signature was:

```
type ConfiguredExtension<E> = Partial<{ title: string; description: string } & E>;

...

    // (Optional) A function that can be used to configure the extension dynamically based on the extension point's context
    configure?: (
      context?: Readonly<Context>
    ) => ConfiguredExtension<ExtraProps> | ConfiguredExtension<ExtraProps>[] | undefined;
```

I'm not sure how the the implementation would look though!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82709. Feel free to continue the discussion around this there. Thank you!
